model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -263,9 +263,27 @@ message StandardPTransforms {
   }
   // Payload for all of these: ParDoPayload containing the user's SDF
   enum SplittableParDoComponents {
+    // Pairs the input element with its initial restriction.
+    // Input: element; output: KV(element, restriction).
     PAIR_WITH_RESTRICTION = 0 [(beam_urn) = "beam:transform:sdf_pair_with_restriction:v1"];
+
+    // Splits the restriction inside an element/restriction pair.
+    // Input: KV(element, restriction); output: KV(element, restriction).
     SPLIT_RESTRICTION = 1 [(beam_urn) = "beam:transform:sdf_split_restriction:v1"];
+
+    // Applies the DoFn to every element/restriction pair in a uniquely keyed
+    // collection, in a splittable fashion.
+    // Input: KV(string, KV(element, restriction)); output: DoFn's output.
+    // The first "string" is an opaque unique key.
```

Comment:
Say that we will specifically be using the bytes coder. This will be needed for Go SDK and other languages which don't which don't have such an easy time passing around arbitrary types.